.github/workflows/generateAutoConfigs.yaml

```diff
@@ -0,0 +1,127 @@
+name: Generate Spring Auto-Configurations
+
+on:
+  workflow_dispatch:
+    inputs:
+      google_cloud_java_commitish:
+        description: 'Commitish or tag of google-cloud-java (e.g. v1.1.0)'
+        required: true
+
+jobs:
+  generateLibraries:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Java 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install Bazel
+        continue-on-error: false
+        run: |
+          set -x
+          sudo apt install apt-transport-https curl gnupg
+          curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor >bazel-archive-keyring.gpg
+          sudo mv bazel-archive-keyring.gpg /usr/share/keyrings
+          echo "deb [arch=amd64 signed-by=/usr/share/keyrings/bazel-archive-keyring.gpg] https://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list
+          sudo apt update && sudo apt install bazel
+      - uses: bazelbuild/setup-bazelisk@v2
+      - name: Create branch
+        id: create_branch_names
+        continue-on-error: false
+        working-directory: spring-cloud-generator
+        run: |
+          set -x
+          NEW_BRANCH_NAME="generated-autoconfig--$(date '+%Y%m%d%H%M%S')"
+          echo "NEW_BRANCH_NAME=$NEW_BRANCH_NAME" >> $GITHUB_OUTPUT
+      - name: Generate library list
+        continue-on-error: false
+        working-directory: spring-cloud-generator
+        run: |
+          set -x
+          bash generate-library-list.sh -c "$GOOGLE_CLOUD_JAVA_COMMITISH"
+        env:
+          GOOGLE_CLOUD_JAVA_COMMITISH: ${{ inputs.google_cloud_java_commitish }}
+      - name: Compile non-autogen libraries
+        continue-on-error: false
+        run: |
+          set -e
+          mvn install -Dcheckstyle.skip -DskipTests
+      - name: Generate libraries
+        continue-on-error: false
+        working-directory: spring-cloud-generator
+        run: |
+          set -x
+          set -e
+          bash generate-all.sh
+      - name: Check for generation errors
+        continue-on-error: false
+        working-directory: spring-cloud-generator
+        run: |
+          set -x
+          if [[ $(ls *failed-library-generations | wc -l) -eq 0 ]]; then
```

Comment:
Should we check something got generated at all?

Reply:
I think the `Detect changes` step would stop the workflow if there were no generated files

Reply:
If nothing got generated in the run (and the modules already exist on main branch), then it might still show up as file changes, just all deletions. That said, I think the `failed-library-generations` check should be catching these, @diegomarquezp  correct me if I’m wrong.

Reply:
Oh, now I see the point @emmileaf and @meltsufin. We have two cases:
1. `generate-library-list.sh` produces an empty list: depending on the branch the WF is ran on, it would either leave an empty `spring-cloud-previews` or whichever modules the source branch already had. Both cases would be catched by `Detect Changes`, which basically counts the files changed in the `spring-cloud-previews` folder (no changes would be made).
2. Every call to `generate-one.sh` fails: If any generation fails, it would be catched in the `failed-library-generations` folder. However, this made me realize that the replacement logic in `generate-one.sh` does not really replace the whole folder (it [uses](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/02267b865b146d9948b33666c72ec6e6d4f16796/spring-cloud-generator/generate-one.sh#L87) the `-o` option in `unzip`), which leaves the possibility of a new version that removes an unnecesary file, for example, but that it's not removed by the script (`unzip -o` will only replace existing files in a folder and will not delete those which are not in the `.zip`). 
That said, adding folder replacement logic to `generate-one.sh` would have a deleted folder still catched since the error detection is basically an exit code capture of `generate-one.sh`, which runs with `set -e` (i.e. the `bazel` call, folder removal and jar extraction are points of failure within the `generate-one.sh` script that are [catched](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/02267b865b146d9948b33666c72ec6e6d4f16796/spring-cloud-generator/generate-all.sh#L36) in the loop of `generate-all.sh`).

Reply:
If `generate-library-list.sh` and/or `generate-all.sh` did absolutely nothing due to a bug, would we catch that as a failure in the CI?

Reply:
I did add a sanity check script/step to confirm start and exit of `generate-all.sh`, `generate-library-list.sh` and `compute-monorepo-tag.sh` (new script).
Basically `touch`es two files at start and exit of each script. The check confirms that all these files have been created